app/models/spree/product.rb

```diff
@@ -257,15 +259,14 @@ def import_date
       variants.map(&:import_date).compact.max
     end
 
-    def destroy
+    def destroy_related_exchange_variants
       transaction do
+        # binding.pry()
         touch_distributors
 
         ExchangeVariant.
           where('exchange_variants.variant_id IN (?)', variants.with_deleted.
           select(:id)).destroy_all
-
-        super
```

Comment:
I'm not sure about this change, it's not doing the same thing as before. Specifically; there's an explicit `transaction` block which was wrapping the whole change, but moving it to a before callback will have different behaviour.

Reply:
Should we just ignore this one then? There isn't a way to preserve the `transaction` block and use a callback I think.

Reply:
Maybe take the commit out of this PR and work on it in a new PR. You can use an `around_destroy` callback to achieve the same logic. Example:
https://github.com/openfoodfoundation/openfoodnetwork/blob/27dcad205e2b0a2aa3dceb608b11ac849c0eadb3/app/models/spree/variant.rb#L240-L243